vendor/processors/stripe_processor.py

```diff
@@ -656,12 +676,19 @@ def update_offers(self, offers):
 
     def sync_offers(self, site):
         products = self.get_site_offers(site)
-        offer_pk_list = [product['metadata']['pk'] for product in products]
+        offer_pk_list = []
+
+        if products:
```

Comment:
remove the empty var init (`offer_pk_list = []`) and the `if products` just make sure you have change `def get_site_offers` to return an empty list instead of None